eth/protocols/eth/receipt.go

```diff
@@ -403,10 +412,20 @@ func (rl *ReceiptList69) EncodeRLP(_w io.Writer) error {
 	return w.Flush()
 }
 
+// ExcludeStateSync removes the state sync transaction from the list. We use
+// the property of zero gas used to identify a state-sync transaction in the
+// list.
+func (rl *ReceiptList69) ExcludeStateSync() {
+	if len(rl.items) > 0 && rl.items[len(rl.items)-1].GasUsed == 0 {
```

Comment:
Since this is the same implementation as ReceiptList68, it could be extracted as a common function and used by both ReceiptList68 and ReceiptList69.

Reply:
Done. There were few more extra changes needed which are also extracted to a common function.